src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -7,118 +7,111 @@
-
-        public override bool IsExtensionMethod => false;
-
-        public override bool HidesBaseMethodsByName => false;
-
-        public override bool IsVararg => false;
-
-        public override bool ReturnsVoid => false;
-
-        public override bool IsAsync => false;
-
-        public override RefKind RefKind => RefKind.None;
-
-        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
-
-        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
-
-        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull => ImmutableHashSet<string>.Empty;
-
-        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations
-            => ImmutableArray<TypeWithAnnotations>.Empty;
-
-        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
-
-        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
-
-        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
-
-        public override Symbol? AssociatedSymbol => null;
-
-        public override Symbol ContainingSymbol => ContainingType;
-
-        public override ImmutableArray<Location> Locations => ContainingType.Locations;
-
-        public override Accessibility DeclaredAccessibility => Accessibility.Public;
-
-        public override bool IsStatic => false;
-
-        public override bool IsVirtual => !IsOverride && !IsAbstract;
-
-        public override bool IsAbstract => ContainingType.IsAbstract;
-
-        public override bool IsSealed => false;
-
-        public override bool IsExtern => false;
 
-        internal override bool HasSpecialName => true;
-
-        internal override MethodImplAttributes ImplementationAttributes => MethodImplAttributes.Managed;
-
-        internal override bool HasDeclarativeSecurity => false;
-
-        internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
-
-        internal override bool RequiresSecurityObject => false;
-
-        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
-
-        internal override bool GenerateDebugInfo => false;
+            if (ContainingType.IsAbstract)
+            {
+                result &= ~DeclarationModifiers.Virtual;
+                result |= DeclarationModifiers.Abstract;
+            }
 
-        public override DllImportData? GetDllImportData() => null;
+            Debug.Assert((result & ~allowedModifiers) == 0);
+            Debug.Assert(modifiersAreValid(result));
+            return result;
 
-        internal override ImmutableArray<string> GetAppliedConditionalSymbols()
-            => ImmutableArray<string>.Empty;
+            static bool modifiersAreValid(DeclarationModifiers modifiers)
```

Comment:
This should probably be marked as conditional. Consider making it a conditional method or filing a follow-up issue (when attributes on local functions become available for use)

Reply:
> This should probably be marked as conditional. Consider making it a conditional method or filing a follow-up issue (when attributes on local functions become available for use)

Added conditional compilation

---
In reply to: [456676274](https://github.com/dotnet/roslyn/pull/46053#discussion_r456676274) [](ancestors = 456676274)